animated-transformer/src/lib/gtensor/gtensor.ts

```diff
@@ -941,14 +949,43 @@ export function makeRange<T extends DName>(
   start: number,
   end: number,
   step: number,
-  dtype: 'float32' | 'int32' = 'float32'
+  dtype: 'float32' | 'int32' = 'float32',
 ): GTensor<T> {
   return new GTensor<T>(tf.range(start, end, step, dtype), [dname]);
 }
 
+/* Returns a 2D triangular mask of shape [size, size]
+ * Parameters:
+ * - size: The dimension of the mask
+ * - dimNames: Names of the dimensions of the final GTensor
+ * - lowerLeftValue : All the values under the main diagonal and including the main diagonal will be replace by this value
+ * - upperRightValue : All the entries avobe the main diagonal of the matrix will be replace by this value
+ * // TODO add optianal broadcastTo dimensions/GTensor
+ * */
+export function makeTriangularMatrix<N extends string, T extends string | number>(
+  size: number,
+  dimNames: N[],
```

Comment:
provide the two names as separate arguments, and they are two different parameters that extend string, otherwise you are inviting runtime errors if someone puts in three names into the dimNames array. 

Also add in the dtype argument (given to the tf.scalar, and the fill).